pkg/frontend/v1/frontend.go

```diff
@@ -256,17 +256,21 @@ func getQuerierID(server frontendv1pb.Frontend_ProcessServer) (string, error) {
 }
 
 func (f *Frontend) queueRequest(ctx context.Context, req *request) error {
-	userID, err := tenant.TenantID(ctx)
+	tenantIDs, err := tenant.TenantIDs(ctx)
 	if err != nil {
 		return err
 	}
 
 	req.enqueueTime = time.Now()
 	req.queueSpan, _ = opentracing.StartSpanFromContext(ctx, "queued")
 
-	maxQueriers := f.limits.MaxQueriersPerUser(userID)
+	// aggregate the max queriers limit in the case of a multi tenant query
+	maxQueriers, err := tenant.AggregatedLimitMinimumOfNonZeroValues(ctx, f.limits.MaxQueriersPerUser)
```

Comment:
`validation` sounds like a better place to me to be honest. How about a `validate.MinimumOfNonZeroValues([]int) (int)` as this would not need the user lookup at all?

Reply:
Sounds good to me.

Reply:
6513f87 should have that now